### ISSUE TITLE
Feature of Disabling Vibrant Colours

### ISSUE THREAD

**Issue by kulkalkul** [KybernetikGames/kybernetikgames.github.io#1]
Hey, I love the work you've done. I was going to email but found the repo of the site, so thought it would be more appropiate than sending an email. Colours of your site are too heavy to my eyes, usually too bright. While some people like that way, I'm unable to concentrate on text due to this. Changing of colours between different projects also affects me (like blue -> green). So, I believe there could be an accessibility button that disables colours, makes it black text over white background. I could write the implementation for it if you don't have the time to spare. Thanks in advance!

A sidenote: I was having issues with animations and saw your comments all over the reddit and unity forums, I considered using animancer lite and maybe upgrading to animancer, but decided not to do so after I saw the documentation colours. So I assume there could be other people like me, who were not able to read documentation and stood away.

**Comment by KybernetikGames** [comment]
I have no real web development experience and no idea where to even begin with something like that, but I'd be happy to work with you to accommodate your needs.

In the mean time, I use the Dark Reader extension for Firefox on sites I don't like the look of. Here's what it can do to the Animancer documentation:
![image](https://user-images.githubusercontent.com/43698611/136291310-a1703019-9114-4138-9066-34f2884cfeaa.png)
Or light mode:
![image](https://user-images.githubusercontent.com/43698611/136291369-deb5cc4a-d976-464d-a479-8648d4446b12.png)
Or greyscale:
![image](https://user-images.githubusercontent.com/43698611/136291502-fa2ac3b6-6c15-46d0-b5be-a455194394b5.png)

**Comment by kulkalkul** [comment]
Wow, neat extension. I have experience though they all are framework-based like React, static websites are too cumbersome to work with, which I did remember right now. 😄

In short, this is what I did:
- Check out if there is a CSS file that can be overriden (couldn't find a single source of truth)
- Thought of reading all styles from CSS and overwriting those (though, new CORS rules that chrome implemented made me regret trying)
- Decided on checking out how DarkReader does all the stuff with CORS ongoing, and figured out there is an website embeddable version of it.

Last one is the easiest way of doing it, here is an example how well it works:
```js
DarkReader.enable({mode: "light", brightness: 210, grayscale: 100});
```

![image](https://user-images.githubusercontent.com/1013994/136392715-d4978bac-3ab3-4947-b4a4-94b5c97eba2f.png)

There are two issues remaining:
- Which interaction enables/disables this? So, there must be a some kind of button that users can interact with. If navbar is consistent all over the site, adding it rightmost of the navbar works.

- Users' choice must persist between loads, so this requires saving it to localStorage or cookies. As this is caused by users' action, it abides GDPR/Cookie-Law as far as I know. But even if there were issues with those laws, as this is simply a documentation website that has low-trafic, I don't think it would create any problems. Though, I'd suggest fact checking it. 😄

So, in summary:
- Adding darkreader's source into assets/js
- Editing all the files and adding button to navbar, importing darkreader, importing the script that handles the work
- The script that allows interaction, saves user's choice to localStorage and enables/disables darkReader according to that choice

Most cumbersome of all is editing all navbars in all files, unfortunately, as this needs to be done to all repos. Though the process can be automated easily. If you are OK with it, I can do the 2 and 3 and send a PR. Only thing you have to do is adding darkreader's source file. I'm not doing that because I feel uncomfortable for adding a thousand line of javascript code to someone elses website. So you downloading that from a source you can trust is more appropiate than me adding it.

**Comment by KybernetikGames** [comment]
My sites are all generated by [Wyam](https://wyam.io/) which just has one nav bar file that gets used for all pages so that wouldn't be an issue. That also means submitting a PR would be pointless because I'd need to make the changes to my source files.

Two things I'm not really happy about are:

- Adding the whole of Dark Reader's source when only a small portion of it would actually get used if it's just a single button.
- Messing around with cookies and GDPR.

Even if you help with the implementation, it still seems like a lot of work to achieve something that's already handled by Dark Reader (and other similar extensions) in a way that gives the user much better control over how they want to see it. And any time I spend on this would be time not spent improving Animancer itself or writing better documentation so I don't think it would be worth it.

For now, I'll try simply reducing the brightness and saturation of the colours and see if I can come up with something that still looks reasonable (though as a programmer I'm not the best judge of that either). I have to go to work shortly, but I'll post some screenshots here when I get a chance to play around with it.

**Comment by KybernetikGames** [comment]
I've darkened the saturation of the main colour and toned down the saturation of the backgrounds:

# Old

![image](https://user-images.githubusercontent.com/43698611/136520985-37dcf47c-dace-44a2-a49a-4168b53ee06f.png)

# New

![image](https://user-images.githubusercontent.com/43698611/136525513-7cfd421b-7cb8-4eea-b51f-1a7ffa60252e.png)

# Old

![image](https://user-images.githubusercontent.com/43698611/136525592-781a2648-3d66-49b7-a1de-ac3e81cf483c.png)

# New

![image](https://user-images.githubusercontent.com/43698611/136525815-57e2ed84-acb3-458f-a813-a5b70dfdf3a5.png)

What do you think?

**Comment by kulkalkul** [comment]
Yeah, I expected some kind of static generator. I wasn't aware of Dark Reader at the time I was writing this, I knew similar extensions but they were only "dark mode only".

Btw, this one is way better than the previous one. 😄 Thanks for our little chat and your work.
